dragontts/app/main.py

```
@@ -88,15 +107,41 @@ async def _checkpoint_loop():
+                    )
+                except ValueError:
+                    elapsed_ok = True
+                if not elapsed_ok:
+                    continue  # window arrived, but the N-day interval hasn't
+            else:
+                await asyncio.sleep(settings.ttl_purge_interval_seconds)
             try:
                 await cache.purge_expired()
+                try:
+                    await metadata.set_meta("last_ttl_purge", str(time.time()))
+                except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)